pkg/util/exec/test/mockexec.go

```diff
@@ -76,3 +80,47 @@ func (e *MockExecutor) ExecuteCommandWithCombinedOutput(command string, arg ...s
 
 	return "", nil
 }
+
+// Mock an executed command with the desired return values.
+// STDERR is returned *before* STDOUT.
+//
+// This will return an error if the given exit code is nonzero. The error return is the primary
+// benefit of using this method.
+//
+// In order for this to work in a `*_test.go` file, you MUST import TestMockExecHelperProcess
+// exactly as shown below:
+//   import exectest "github.com/rook/rook/pkg/util/exec/test"
+//   // import TestMockExecHelperProcess
+//   func TestMockExecHelperProcess(t *testing.T) {
+//   	exectest.TestMockExecHelperProcess(t)
+//   }
+// Inspired by: https://github.com/golang/go/blob/master/src/os/exec/exec_test.go
+func MockExecCommandReturns(t *testing.T, stdout, stderr string, retcode int) error {
+	cmd := exec.Command(os.Args[0], "-test.run=TestMockExecHelperProcess") //nolint:gosec //Rook controls the input to the exec arguments
+	cmd.Env = append(os.Environ(),
+		"GO_WANT_HELPER_PROCESS=1",
+		fmt.Sprintf("GO_HELPER_PROCESS_STDOUT=%s", stdout),
+		fmt.Sprintf("GO_HELPER_PROCESS_STDERR=%s", stderr),
+		fmt.Sprintf("GO_HELPER_PROCESS_RETCODE=%d", retcode),
+	)
+	err := cmd.Run()
+	return err
+}
+
+// TestHelperProcess isn't a real test. It's used as a helper process for MockExecCommandReturns to
+// simulate output from a command. Notably, this can return a realistic os/exec error.
+// Inspired by: https://github.com/golang/go/blob/master/src/os/exec/exec_test.go
+func TestMockExecHelperProcess(t *testing.T) {
+	if os.Getenv("GO_WANT_HELPER_PROCESS") != "1" {
+		return
+	}
+
+	// test should set these in its environment to control the output of the test commands
+	fmt.Fprint(os.Stderr, os.Getenv("GO_HELPER_PROCESS_STDERR")) // return stderr before stdout
+	fmt.Fprint(os.Stdout, os.Getenv("GO_HELPER_PROCESS_STDOUT"))
+	rc, err := strconv.Atoi(os.Getenv("GO_HELPER_PROCESS_RETCODE"))
+	if err != nil {
+		panic(err)
+	}
+	os.Exit(rc)
+}
```

Comment:
We want to specifically get an error with a parsable return code from the new unit tests, so I move some code from `exec/exec_test.go` that I used for the same purpose to `exec/test/mockexec.go` to allow for reuse.